src/System.IO.FileSystem/tests/Directory/Delete.cs

```diff
@@ -204,6 +205,17 @@ public void UnixShouldBeAbleToDeleteHiddenDirectory()
             Assert.False(Directory.Exists(testDir));
         }
 
+        [Fact]
+        [OuterLoop("Needs sudo access")]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        public void Unix_NotFoundDirectory_ReadOnlyVolume()
+        {
+            ReadOnly_FileSystemHelper(readOnlyDirectory =>
+            {
+                Assert.Throws<DirectoryNotFoundException>(() => Delete(Path.Combine(readOnlyDirectory, "DoesNotExist")));
```

Comment:
A '/' should probably be added at the end of the path, just so that it's clear in the path that this is a directory and not a file.